frontend/app/routes/protected/person-case/route-helpers.server.ts

```diff
@@ -0,0 +1,58 @@
+import type { Actor } from 'xstate';
+
+import { LogFactory } from '~/.server/logging';
+import { i18nRedirect } from '~/.server/utils/route-utils';
+import type { I18nRouteFile } from '~/i18n-routes';
+import { loadMachineActor } from '~/routes/protected/person-case/state-machine.server';
+import type { Machine, StateName } from '~/routes/protected/person-case/state-machine.server';
+
+const log = LogFactory.getLogger(import.meta.url);
+
+/**
+ * Arguments for loadMachineContextOrRedirect function
+ */
+type LoadMachineContextOrRedirectArgs = {
+  /** The current application session */
+  session: AppSession;
+  /** The incoming request */
+  request: Request;
+  /** Optional state name to validate */
+  stateName?: StateName;
+  /** Route to redirect to if tab ID or machine actor not found */
+  i18nRedirectRouteFile?: I18nRouteFile;
+};
+
+type LoadedMachineContext = { machineActor: Actor<Machine>; tabId: string };
+
+/**
+ * Loads the necessary machine context (actor and tab ID) from the request or redirects if not found.
+ * This function ensures all required components for state machine operations are available before proceeding.
+ *
+ * @returns Object containing the machine actor and tab ID
+ * @throws Redirects to the specified route if tab ID is missing or machine actor cannot be loaded
+ */
+export function loadMachineContextOrRedirect({
+  session,
+  request,
+  stateName,
+  i18nRedirectRouteFile = 'routes/protected/index.tsx',
+}: LoadMachineContextOrRedirectArgs): LoadedMachineContext {
```

Comment:
I think the better approach here would be to do:

```ts
function (session: AppSession, request: Request, opts?: Options);
```

This follows the convention where required parameters are explicit and are excluded from the `opts` object (which contains behavior-changing options).

Reply:
Good idea. I will fix it.

Reply:
Refactored